packages/protocol/test/governance/downtime_slasher_intervals.ts

```diff
@@ -0,0 +1,731 @@
+              await presetParentSealForBlocks(startSlots[i], endSlots[i] - startSlots[i] + 1, [
+                bitmapWithoutValidator[validatorIndexInEpoch],
+              ])
+            }
+            await generateProofs(startSlots, endSlots)
+            await slasher.slash(
+              startSlots,
+              endSlots,
+              [validatorIndexInEpoch],
+              0,
+              [],
+              [],
+              [],
+              [],
+              [],
+              []
+            )
+            const balance = await mockLockedGold.accountTotalLockedGold(validatorList[0])
+            assert.equal(balance.toNumber(), 40000)
+          })
+          it('fails if endSlot(i) is not between [startSlot(i+1)-1, endSlot(i+1)]', async () => {
+            const startSlots = [
+              startBlock,
+              startBlock + intervalSize * 2,
+              startBlock + intervalSize,
+            ]
+            const endSlots = [
+              startSlots[0] + intervalSize - 1,
+              startSlots[1] + intervalSize - 1,
+              startBlock + slashableDowntime - 1,
+            ]
+            // the window is covered with slot(0) and slot(2), but it breaks the "chain"
+
+            for (let i = 0; i < startSlots.length; i += 1) {
+              await presetParentSealForBlocks(startSlots[i], endSlots[i] - startSlots[i] + 1, [
+                bitmapWithoutValidator[validatorIndexInEpoch],
+              ])
+            }
+            await generateProofs(startSlots, endSlots)
+            await assertRevert(
+              slasher.slash(
+                startSlots,
+                endSlots,
+                [validatorIndexInEpoch],
+                0,
+                [],
+                [],
+                [],
+                [],
+                [],
+                []
+              ),
+              'at least one endBlock is not in the boundaries of the next interval'
+            )
+          })
+        })
+        it("fails if the slots don't cover the SlashableDowntime window", async () => {
+          const startSlots = [startBlock, startBlock + intervalSize]
+          const endSlots = [
+            startSlots[0] + intervalSize - 1,
+            startSlots[1] + intervalSize - 1,
+            startBlock + slashableDowntime - 1,
+          ]
+          for (let i = 0; i < startSlots.length; i += 1) {
+            await presetParentSealForBlocks(startSlots[i], endSlots[i] - startSlots[i] + 1, [
+              bitmapWithoutValidator[validatorIndexInEpoch],
+            ])
+          }
+          await generateProofs(startSlots, endSlots)
+          await assertRevert(
+            slasher.slash(startSlots, endSlots, [validatorIndexInEpoch], 0, [], [], [], [], [], []),
+            'the intervals are not covering the slashableDowntime window'
+          )
+        })
+      })
+      describe('when succeds', () => {
+        let resp: any
```

Comment:
Question: does Truffle expose a type for transaction receipts?

Reply:
Yes, `Truffle.TransactionResponse`. Adding it